### DISCUSSION TITLE
Does uploading a backup to the cloud create a local tar file?

### DISCUSSION THREAD

**Discussion by rodrigargar** [Altinity/clickhouse-backup#152] [Q&A] [answered]
As I understood from the readme and the code (I am pretty new to _go_), clickhouse-backup does **not** create a tar file locally when uploading a backup to the cloud.

Instead, clickhouse-backup takes the content of a particular backup folder and streams it to the cloud while compressing the data on the fly. Therefore, the tar file is only created in the cloud storage destination.

Is my understanding correct?

**Comment by AlexAkulov** [comment, marked as answer]
Hi @rodrigargar!
Yes, you understood correctly.
Locally backup largely consists of hardlinks and this format is most of efficiently. In fact you store only differences between current version of data and version in backup.
Unfortunately remote storages is not supported hardlinks and we can't use the same way for store remote backups. 
Backup can contain many files, and it has been experimentally found that uploading/downloading large numbers of files in tar is faster, so tar is used.
